File: LeetCode/sumTwoIntegers.cpp
#include <bits.h>
#include <iostream>

class Solution
{
public:
    int getSum(int a, int b)
    {
        while (b != 0)
        {
            int temp = (a & b) << 1;
            a = a ^ b;
            b = temp;
        }
        return a;
    }
};

int main()
{
    Solution sol = Solution();
    std::cout << sol.getSum(1000, -1000) << std::endl;
    return 0;
}

File: LeetCode/sum_two_integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        # No carry
        while b != 0:

            temp = (a & b) << 1
            # print(bin(temp), bin(b))
            a = a ^ b
            # print(bin(a))
            # Check for carry bit
            b = temp
            # print(bin(b))

        # print("\n", bin(a))
        return a

sol = Solution()
print(sol.getSum(-1,1))
